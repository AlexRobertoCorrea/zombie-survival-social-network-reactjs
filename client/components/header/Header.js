import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { observer } from 'mobx-react';
import AddIcon from '@material-ui/icons/Add';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  root: {
    flexGrow: 1,
    padding: theme.spacing.unit * 2
  },
  leftIcon: {
    marginRight: theme.spacing.unit
  }
});

@observer class Header extends Component {
  render() {
    const { classes } = this.props;
    
    return (
      <Grid
        container
        spacing={16}
        justify="space-around"
        alignItems="center"
        className={classes.root}
      >
        <Grid item>
          <Typography
            variant="display2"
            align="center"
          >
            Survivor's list
          </Typography>
        </Grid>
    
        <Grid item>
          <Button
            variant="extendedFab"
            color="primary"
            component="span"
          >
            <AddIcon className={classes.leftIcon} />
            Add survivors
          </Button>
        </Grid>
      </Grid>
    );
  }
}

Header.propTypes = {
  classes: PropTypes.object.isRequired
};


export default observer(withStyles(styles)(Header));
