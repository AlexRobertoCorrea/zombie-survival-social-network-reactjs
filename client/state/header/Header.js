import httpStatus from 'http-status';

const validateForm = async (form) => {
  const { isValid } = await form.validate({ showErrors: true });
  
  return isValid;
};

const fetchData = async (store) => {
  await store.peopleStore.fetchPeople();
};

const savePersonData = async (person, store) => await store.peopleStore.createPerson(person);

const successSave = async (status, person, store, notifier) => {
  if (status === httpStatus.CREATED) {
    notifier.update({
      message: `${person.name} was created successfully`,
      variant: 'success'
    });
  }
  
  return await fetchData(store);
};

const errorSave = (notifier, person, status) => {
  if (status === httpStatus.UNPROCESSABLE_ENTITY) {
    notifier.update({
      message: `${person.name} was already created`,
      variant: 'info'
    });
  } else {
    notifier.update({
      message: `Error on saving ${person.name}. Contact the administrator for more information.`,
      variant: 'error'
    });
  }
};

const saveData = async (person, store, notifier) => {
  try {
    const { status } = await savePersonData(person, store);
    
    return await successSave(status, person, store, notifier);
  } catch (err) {
    const status = err.response.status;
    
    errorSave(notifier, person, status);
  
    return status;
  }
};

const getItems = (person) => {
  let items = '';
  
  if (person.water) {
    items += `Water:${person.water};`;
  }
  
  if (person.food) {
    items += `Food:${person.food};`;
  }
  
  if (person.medication) {
    items += `Medication:${person.medication};`;
  }
  
  if (person.ammunition) {
    items += `Ammunition:${person.ammunition}`;
  }
  
  return items;
};

const getLonlat = person => `Point(${person.latitude} ${person.longitude})`;

const excludePersonValues = (person) => {
  delete person.latitude;
  delete person.longitude;
  delete person.water;
  delete person.food;
  delete person.medication;
  delete person.ammunition;
  
  return person;
};

const getPersonData = (form) => {
  const person = form.values();
  
  person.items = getItems(person);
  person.lonlat = getLonlat(person);
  person.age = Number(person.age);
  
  return excludePersonValues(person);
};

const handleSavePerson = async ({
  form,
  store,
  notifier,
  handleCloseDialogSurvivor
}) => {
  const isValid = await validateForm(form);

  if (isValid) {
    handleCloseDialogSurvivor();

    const person = getPersonData(form);

    return await saveData(person, store, notifier, handleCloseDialogSurvivor);
  }
};

export {
  handleSavePerson
};
