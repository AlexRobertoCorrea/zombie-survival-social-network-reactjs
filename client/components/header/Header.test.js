import React from 'react';

import Header from './Header';

describe('<Header />', () => {
  describe('Renders header correctly', () => {
    const header = shallow(<Header />);
    
    it('Ensure that header works correctly', () => {
      expect(header).toMatchSnapshot();
    });
  });
});
