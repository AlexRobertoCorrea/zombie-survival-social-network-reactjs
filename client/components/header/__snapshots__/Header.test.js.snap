// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Renders header correctly Ensure that header works correctly 1`] = `
<Header
  classes={
    Object {
      "leftIcon": "Header-leftIcon-02",
      "root": "Header-root-01",
    }
  }
/>
`;
